### PR TITLE
fix: log trace should not log sql values

### PR DESCRIPTION
**Pull request by ssoroka**
### What did this pull request do?

Log parameterized queries instead of full sql queries that may contain sensitive information

### User Case Description

See https://github.com/go-gorm/gorm/issues/5287

Closes https://github.com/go-gorm/gorm/issues/5287

### CODE REVIEW

**Comment by jinzhu** [comment]
Hello @ssoroka 

Thank you for your PR, but I think we can change log's callback method to 

`func(... logger.Config) (string, int64)` and use the Config param to control returns parameterized or full SQL.

Can you make the change?

**Comment by dorsha** [comment]
Hi, any update with this one? 
It is really important for us (and I think any consumer), since PII is getting leaked when enabling sql logs.
Any help needed to close this one?

**Comment by finnnark** [comment]
+1

**Comment by ssoroka** [comment]
PR Updated. let me know if this is what you were thinking. Test failures don't seem related to my change.

File: callbacks.go [hunk 1]
```diff
@@ -131,8 +132,12 @@ func (p *processor) Execute(db *DB) *DB {
 	}
 
 	if stmt.SQL.Len() > 0 {
-		db.Logger.Trace(stmt.Context, curTime, func() (string, int64) {
-			return db.Dialector.Explain(stmt.SQL.String(), stmt.Vars...), db.RowsAffected
+		db.Logger.Trace(stmt.Context, curTime, func(config ...logger.Config) (string, int64) {
+			if len(config) > 0 && config[0].ParameterizedQueries {
+				return stmt.SQL.String(), db.RowsAffected
+			} else {
```
Comment (demoManito) [inline callbacks.go:138]: Can else be deleted here?

**Comment by dageev-hs** [comment]
Hi @ssoroka @demoManito 
Any updates on this PR?

**Comment by thundering-herd** [comment]
Hej

Is there any way to support this to be integrated into `gorm`? @ssoroka @demoManito

**Pull request closed without merging**